### PR TITLE
Provide ability to use native S3 client (via S3 Transfer Manager) to speed-up uploads/downloads

### PR DESCRIPTION
**Pull request by gafiatulin**
- feat: Provide ability to use native S3 client (via S3 Transfer Manager) to speed-up uploads/downloads.
- refactor: builder interface for *Store.
- fix: prefer show interpolator everywhere

### CODE REVIEW

File: azure/src/main/scala/blobstore/azure/AzureStore.scala [hunk 1]
```diff
@@ -223,37 +226,69 @@ class AzureStore[F[_]: Async](
...
+  trait AzureStoreBuilder[F[_]] {
+    def withClient(client: BlobServiceAsyncClient): AzureStoreBuilder[F]
+    def withQueueSize(queueSize: Int): AzureStoreBuilder[F]
+    def withBlockSize(blockSize: Int): AzureStoreBuilder[F]
+    def withNumBuffers(numBuffers: Int): AzureStoreBuilder[F]
+    def enableFullMetadata: AzureStoreBuilder[F]
+    def disableFullMetadata: AzureStoreBuilder[F]
+    def enableTrailingSlashFiles: AzureStoreBuilder[F]
+    def disableTrailingSlashFiles: AzureStoreBuilder[F]
+    def build(): ValidatedNec[Throwable, AzureStore[F]]
```
Comment (jgogstad) [inline azure/src/main/scala/blobstore/azure/AzureStore.scala:243]: Drop empty paren for consistency maybe

File: s3/src/main/scala/blobstore/s3/S3Store.scala [hunk 2]
```diff
@@ -477,14 +462,98 @@ object S3Store {
...
+    _crtClient: Option[S3AsyncClient] = None,
+    _objectAcl: Option[ObjectCannedACL] = None,
+    _sseAlgorithm: Option[ServerSideEncryption] = None,
+    _defaultFullMetadata: Boolean = false,
+    _defaultTrailingSlashFiles: Boolean = false,
+    _bufferSize: Int = S3Store.multiUploadDefaultPartSize.toInt,
+    _queueSize: Int = 32
+  ) extends S3StoreBuilder[F] {
+    def withS3Client(s3Client: S3AsyncClient): S3StoreBuilder[F]               = this.copy(_s3Client = s3Client)
+    def setCrtClient(maybeCrtClient: Option[S3AsyncClient]): S3StoreBuilder[F] = this.copy(_crtClient = maybeCrtClient)
```
Comment (jgogstad) [inline s3/src/main/scala/blobstore/s3/S3Store.scala:503]: Can we take `S3AsyncClient with S3CrtAsyncClient` and avoid that runtime type check?

**Review by jgogstad — APPROVED** [review]
LGTM, see question

File: s3/src/main/scala/blobstore/s3/S3Store.scala (hunk 2 above)
Comment (gafiatulin) [reply s3/src/main/scala/blobstore/s3/S3Store.scala:503]: The reason I'm not referencing `S3CrtAsyncClient` by type and doing runtime reflection instead is that I didn't want to require preview dependency (`software.amazon.awssdk:s3-transfer-manager`) that transitively brings in CRT (10MB Jar on it's own) for an optional feature.

**Pull request merged**